deployment/configenv/xml_jlibpt/ComponentBase.cpp

```diff
@@ -0,0 +1,384 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2018 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#include "ComponentBase.hpp"
+#include "deployutils.hpp"
+#include "configenvhelper.hpp"
+#include "buildset.hpp"
+
+namespace ech
+{
+
+ComponentBase::ComponentBase(const char* name, EnvHelper * envHelper)
+{
+  this->name.clear().appendf("%s",name);
+  assert(envHelper);
+  this->envHelper = envHelper;
```

Comment:
To distinguish member variable with input parameter.

Reply:
Our normal convention when passing in a value to be assigned to a member variable (to avoid a name clash) is to add a _ on the front of the parameter name.